daemon/api_users.go

```diff
@@ -408,6 +408,12 @@ func getUserDetailsFromStore(theStore snapstate.StoreService, email string) (str
 		return "", nil, fmt.Errorf("cannot create user for %q: no ssh keys found", email)
 	}
 
+	// Amend information where the key came from to ensure it can
+	// be update/replaced later
+	for i, k := range v.SSHKeys {
+		v.SSHKeys[i] = fmt.Sprintf(`%s # {"origin":"store","email":%q}`, k, email)
```

Comment:
We should probably aim for something like
```
# snapd: {"origin":"store","email":"foo@bar.baz"}
<actual key>
```

Reply:
But then we risk that if the user manually deletes lines from `~/.ssh/authorized_keys` the comment will be accidentally left over, and apply to the next line below it.

It's also true that the user might also edit the key line if our comment is at the end of it, but that's way less likely to happen or to cause trouble if it happens.